### ISSUE TITLE
Provide better `gpu` configuration information at runtime

### ISSUE THREAD

**Issue by Helveg** [arbor-sim/arbor#1861]
**Describe the feature you need**
One needs to pass a `--gpu` flag to `ABC`. In most (/all?) cases this needs to match whatever `arbor` module can be found in the active env, so instead we could probe if an `arbor` can be found and which `gpu` settings were used to build it. To this end we'd require the `arbor.config()["gpu"]` tag to be improved to reflect the gpu settings.

**Explain what it is supposed to enable**
Tools like ABC can use the imporved arbor gpu configuration info to build with proper gpu support, without the user having to specify this.

**Comment by thorstenhater** [comment]
Hi @Helveg,

I think that's not entirely correct: The catalogue can be built with GPU support and used in a 
simulation linking a CPU-only libarbor. So, in theory, such a catalogue can support multiple
versions of libarbor as long as it supports a superset of all their features.

**Comment by Helveg** [comment]
It's about having to say `gpu=cuda`, while arbor already knows it will only accept `cuda`, so why not expose this value so ABC can pick it up

**Comment by thorstenhater** [comment]
That's also not entirely correct ;) We have at least one system where one head node manages multiple backend,
some without GPU, some with NV GPUs and some with AMD GPUs.

**Comment by Helveg** [comment]
Right, then the value in `arbor.config()` can be an array of supported backends, and `--gpu` in ABC turns on all supported backends?

**Comment by thorstenhater** [comment]
Ah, here comes an ugly truth: We can only ever support 
- one CPU backend (arch+SIMD=0/1)
- one GPU backend (HIP/CUDA)

so the a-b-c tool needs to be coupled to one libarbor. But. You could
totally use the CPU only cat with a GPU lib (as long as you never ask
for a GPU mech). And vice-versa, using the GPU cat with a CPU lib
is also fine, as we never enter the GPU parts. 

So, you could use the CUDA-GPU-cat with a HIP-GPU-lib (and the other
way), if
1. the CPU architectures match
2. you never ask to enter a GPU kernel

Not saying that's likely, but possible ;)

**Comment by Helveg** [comment]
Well, I'm pretty sure that's more of a bug than a feature, since it'll error if you actually use the GPU parts, and we can safely assume, as a same default, that if you have a CUDA GPU libarbor, and you don't specify anything, that you want to support CUDA when you say "turn on (my cuda) GPU support" when you lazily give "--gpu". Or, why I'm actually asking this: when you don't even know the GPU kind of your arbor/hardware on HOC and you don't want to bother finding out ;p
